docs/source/profiling.rst

```diff
@@ -0,0 +1,498 @@
+Profiling coupled simulations
+=============================
+
+The first challenge one faces when creating a coupled simulation is getting it
+to run correctly. Once the model is running and producing results however, the
+question often becomes how to make it run faster, so as to be able to do science
+more efficiently. The model needs to be optimised.
+
+The first step of optimisation is measurement: how much compute time does the
+model spend on what? Without measurement, you end up optimising parts that don't
+contribute significantly to the total run time, thus making little progress.
+
+Measuring and optimising individual submodels is outside of the scope of
+MUSCLE3. This should be done at the level of the individual model, using
+profilers like gprof or perf or TotalView, and there isn't much MUSCLE3 can
+contribute. In a coupled simulation however, you first need to know which of
+the coupled models takes the most compute time and should therefore be
+optimised, and this MUSCLE3 can help with.
+
+A MUSCLE3 simulation consists of a set of submodels, which during the simulation
+spend some of their time computing, some of it waiting for inputs to arrive from
+other submodels, and some of it transferring data. The communication activities
+are done through ``libmuscle``, and computing (and internal communication, if
+the model is parallelised) takes place in between them. During each run,
+``libmuscle`` will measure when each external communication takes place, how
+much time was spent waiting, and how much time was spent transferring the data.
+This information is collected centrally and saved to a file, which you can then
+analyse to discover where the coupled simulation can best be optimised.
+
+Configuring profiling
+---------------------
+
+Profiling is enabled by default, so you do not have to do anything to get
+MUSCLE3 to produce profiling data. For typical coupled simulations consisting of
+only a few instances that do not communicate very often, this should not affect
+the simulation performance. If you have a simulation with very many instances
+which communicate intensively, then the overhead of sending all that information
+to the central database can become significant.
+
+If that is the case, then you should do your profiling with a smaller number of
+instances, and disable profiling for the large run to avoid the overhead.
+Profiling can be disabled by setting the setting ``muscle_profile_level`` to
```

Comment:
Might be good to include an overview of `special` settings somewhere in the documentation. Not for this PR (you may create an issue if you agree). The other special settings I can think of is the remote log level, but perhaps there are more already.

Reply:
I think it's only these two so far, but good idea. #235